src/uct/ib/rc/accel/rc_mlx5_common.h

```diff
@@ -514,4 +514,75 @@ uct_rc_mlx5_txqp_dptr_post(uct_rc_iface_t *iface, uct_rc_txqp_t *txqp, uct_ib_ml
                                opmod, signal, wqe_size, av, qp_type);
 }
 
+static UCS_F_ALWAYS_INLINE
+void uct_rc_mlx5_txqp_dptr_post_iov(uct_rc_iface_t *iface, uct_rc_txqp_t *txqp,
+                                    uct_ib_mlx5_txwq_t *txwq, unsigned opcode_flags,
+                         /* IOV  */ const uct_iov_t *iov, size_t iovcnt,
+                         /* SEND */ uint8_t am_id, const void *am_hdr, unsigned am_hdr_len,
+                         /* RDMA */ uint64_t remote_addr, uct_rkey_t rkey,
+                         /* AV   */ uct_ib_mlx5_base_av_t *av,
+                                    int signal, int qp_type)
+{
+    struct mlx5_wqe_ctrl_seg     *ctrl;
+    struct mlx5_wqe_raddr_seg    *raddr;
+    struct mlx5_wqe_data_seg     *dptr;
+    struct mlx5_wqe_inl_data_seg *inl;
+    uct_rc_hdr_t                 *rch;
+    unsigned                      wqe_size, inl_seg_size, ctrl_seg_size;
+
+    if (!signal) {
+        signal = uct_rc_iface_tx_moderation(iface, txqp, MLX5_WQE_CTRL_CQ_UPDATE);
+    } else {
+        ucs_assert(signal == MLX5_WQE_CTRL_CQ_UPDATE);
+    }
+
+    ctrl = txwq->curr;
+    ctrl_seg_size = uct_ib_mlx5_ctrl_seg_size(qp_type);
+
+    switch (opcode_flags) {
+    case MLX5_OPCODE_SEND:
+        inl_seg_size     = ucs_align_up_pow2(sizeof(*inl) + sizeof(*rch) + am_hdr_len,
+                                             UCT_IB_MLX5_WQE_SEG_SIZE);
+
+        ucs_assert(ctrl_seg_size + inl_seg_size + sizeof(*dptr) <=
```

Comment:
this assertion assumes 1 iov entry. need to move this assertion after calling uct_ib_mlx5_set_data_seg_iov() because only then we know the real length.